plugin/trino-password-authenticators/src/main/java/io/trino/plugin/password/ldap/JdkLdapAuthenticatorClient.java

```diff
@@ -119,6 +120,21 @@ public boolean isGroupMember(String searchBase, String groupSearch, String conte
         }
     }
 
+    @Override
+    public Set<String> lookupUserAttributeValues(String searchBase, String searchFilter, String contextUserDistinguishedName, String contextPassword, String attributeName) throws NamingException {
```

Comment:
Can we extract this as a separate method - which would be re-used by `isGroupMember` and `lookupDistinguisedName` ? 

Like we could have something like this 

```
    public <T> T unknownMethod(String searchBase, String searchFilter, String contextUserDistinguishedName, String contextPassword, Function<CloseableSearchResults, T> function)
            throws NamingException
    {
        try (CloseableContext context = createUserDirContext(contextUserDistinguishedName, contextPassword);
                CloseableSearchResults search = searchContext(searchBase, searchFilter, context)) {
            return function.apply(search);
        }
        
    }
    ````
    
    This would help us for LdapGroupProvider - we could additionally pass the attributes which we require

Reply:
done